packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -94,25 +81,44 @@ export function BlockEditor( {
 
 	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
 		'postType',
-		productType,
+		postType,
 		{ id: productId }
 	);
 
 	const { updateEditorSettings } = useDispatch( 'core/editor' );
 
 	useLayoutEffect( () => {
-		const template = settings?.templates?.[ productType ];
+		const productTemplates = settings?.productTemplates ?? [];
+		const productTemplate = productTemplates.find(
+			( template ) => template.product_data.type === productType
+		);
```

Comment:
Is this just until we store the product template ID with the product?

Reply:
Correct! For now we don't have another way until we have the relation.